tests/Tests/Axis3d.elm

```diff
@@ -133,6 +136,65 @@ intersectionWithPlane =
         )
 
 
+intersectionWithTriangle : Test
+intersectionWithTriangle =
+    Test.fuzz3
+        Fuzz.triangle3d
+        Fuzz.direction3d
+        Fuzz.vector3d
+        "intersectionWithTriangle works properly"
+        (\triangle axisDirection parameters ->
+            let
+                ( p0, p1, p2 ) =
+                    Triangle3d.vertices triangle
+
+                e1 =
+                    Vector3d.from p0 p1
+
+                e2 =
+                    Vector3d.from p0 p2
+
+                ( u, v, t ) =
+                    Vector3d.toTuple Quantity.unwrap parameters
+
+                planeIntersection =
+                    p0
+                        |> Point3d.translateBy (Vector3d.scaleBy u e1)
+                        |> Point3d.translateBy (Vector3d.scaleBy v e2)
+
+                direction =
+                    Vector3d.withLength (Length.meters 1) axisDirection
+
+                axisOrigin =
+                    planeIntersection
+                        |> Point3d.translateBy (Vector3d.scaleBy -t direction)
+
+                maybeAxis =
+                    Axis3d.throughPoints axisOrigin planeIntersection
```

Comment:
It looks to me like you could avoid the `Maybe` here by using `Axis3d.through axisOrigin direction` (or equivalently `Axis3d.withDirection direction axisOrigin`)...

Reply:
Done...